sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -616,17 +597,22 @@ sycl::detail::pi::PiProgram ProgramManager::getBuiltPIProgram(
   if (auto exception = checkDevSupportDeviceRequirements(Device, Img))
     throw *exception;
 
-  auto BuildF = [this, &Img, &Context, &ContextImpl, &Device, Prg, &CompileOpts,
+  auto BuildF = [this, &Img, &Context, &ContextImpl, &Device, &CompileOpts,
                  &LinkOpts, SpecConsts] {
     const PluginPtr &Plugin = ContextImpl->getPlugin();
-    applyOptionsFromImage(CompileOpts, LinkOpts, Img, {Device}, Plugin);
+    std::string CompileOptsString =
+        CompileOpts.empty() ? "" : std::string(CompileOpts);
+    std::string LinkOptsString = LinkOpts.empty() ? "" : std::string(LinkOpts);
+    applyOptionsFromImage(CompileOptsString, LinkOptsString, Img, {Device},
+                          Plugin);
+    std::string_view CompileOptsupdated(CompileOptsString);
+    std::string_view LinkOptsupdated(LinkOptsString);
```

Comment:
Since we're creating strings here anyway, there's no benefit to using `string_view` when we can just pass around const references to string. So `getOrCreatePIProgram`, `build` and `putItemToDisc` should probably keep accepting `const std::string &`.